pkg/asset/machines/aws/machines.go

```diff
@@ -170,11 +170,30 @@ func Machines(clusterID string, region string, subnets map[string]string, pool *
 	return machines, controlPlaneMachineSet, nil
 }
 
-func provider(in *machineProviderInput) (*machineapi.AWSMachineProviderConfig, error) {
+func provider(in *machineProviderInput, pool *aws.MachinePool) (*machineapi.AWSMachineProviderConfig, error) {
 	tags, err := tagsFromUserTags(in.clusterID, in.userTags)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to create machineapi.TagSpecifications from UserTags")
 	}
+
+	securityGroups := []machineapi.AWSResourceReference{{
+		Filters: []machineapi.Filter{{
+			Name:   "tag:Name",
+			Values: []string{fmt.Sprintf("%s-%s-sg", in.clusterID, in.role)},
+		}},
+	}}
+	securityGroupsFromPool := []machineapi.AWSResourceReference{}
+	for _, sgID := range pool.AdditionalSecurityGroupIDs {
+		tmpID := sgID
```

Comment:
Nit: btw, this is also valid and makes the linter happy: 
```suggestion
		sgID := sgID
```
it creates a new `sgID` variable with a different scope therefore it's safe to take its address.